MachObfuscator/HeadersParsing/ObjectSymbolsLoader.swift

```diff
@@ -0,0 +1,5 @@
+import Foundation
+
+protocol ObjectSymbolsLoader {
```

Comment:
You are totally right, I made it so wrong! `SimpleSourceSymbolsLoader` can only load source files browsing the directory recursively. `TextFileSymbolListLoader` can only load new-line-separated lists.
What do you think about splitting this protocol as two independent protocols? Do you have better suggestion?

Reply:
Actually, there are 3 implementations of `ObjectSymbolsLoader` - there is also one which loads MachO image.

It depends what is really needed. Currently I understand that the protocol is needed because unit tests need to pass mock implementations instead of real ones (at least for version which loads MachO image). In the future protocol may enable adding other features like caching (similar to decorator design pattern).

If we want to address only the first need, simplest solution would be to create one protocol (eg. `ObjectSymbolsLoader`) with 3 methods: `loadFromImage`, 'loadFromSources` and `loadFromTextFile` which can simply be mocked (BTW, this would be a bit similar to factory design pattern) 

Probably better solution would be to define 3 protocols or replace these single-method protocol with functions/closures (something like https://www.swiftbysundell.com/articles/mock-free-unit-tests-in-swift/#functional-behavior) which probably would be simpler, but still would provide possibility to mock and decorate them. To make them explicit about what they load, `URL` could be wrapped as for example `TextFileURL` and `SourceFilesURL`.

If we want to implement different formats for symbol lists (text, JSON, XML etc) then maybe it is not needed to separate URL storage from the loading logic. It complicates code but is used only to handle command line arguments.